src/transformers/pipelines/text_to_audio.py

```diff
@@ -190,76 +242,182 @@ def _forward(self, model_inputs, **kwargs):
+            elif "sequences" in audio:  # E.g Dia (these models will need the processor to postprocess)
+                waveform = audio["sequences"]
             else:
-                waveform = audio
-        # Or we need to postprocess to get the waveform
+                raise ValueError(
+                    f"Unexpected keys in the audio output format: {audio.keys()}. Expected one of "
+                    "`waveform` or `audio`"
+                )
+        elif isinstance(audio, (tuple, list)):
+            waveform = audio[0]
+        else:
+            waveform = audio
+
+        # If the data is a LongTensor, then it is a codebook that needs to be decoded. If the model is not doing the
+        # decoding itself, then it's because the decoding happens in the processor.
+        if isinstance(waveform, torch.LongTensor):
+            waveform = self.processor.decode(waveform)
+
+        # If we know there is only one audio channel, we can infer missing dimensions
+        if self.audio_channels == 1:
+            if len(waveform.shape) == 1:  # (audio_length) -> (bsz=1, audio_length)
+                waveform = waveform.unsqueeze(0)
+            if len(waveform.shape) == 2:  # (bsz, audio_length) -> (bsz, audio_channels=1, audio_length)
+                waveform = waveform.unsqueeze(1)
+
+        # The waveform MUST have shape (batch_size, audio_channels, audio_length) at this point
+        if len(waveform.shape) != 3:
+            raise ValueError(
+                f"Unexpected waveform shape: {waveform.shape}. Expected (batch_size, audio_channels, audio_length)"
+            )
+
+        # bsz == 1 -> output a single dict
+        if waveform.shape[0] == 1:
+            output_dict["audio"] = waveform[0].to(device="cpu", dtype=torch.float).numpy()
+            output_dict["sampling_rate"] = self.sampling_rate
+            return output_dict
+        # bsz > 1 -> output a list of dicts
         else:
-            waveform = self.processor.decode(audio)
+            output_list = []
+            for batch_idx in range(waveform.shape[0]):
+                output_dict = {}
+                output_dict["audio"] = waveform[batch_idx].to(device="cpu", dtype=torch.float).numpy()
+                output_dict["sampling_rate"] = self.sampling_rate
+                output_list.append(output_dict)
+            return output_list
+
+    def save_audio(self, audio: Union[dict[str, Any], list[dict[str, Any]]], path: Union[str, list[str]]):
+        """
+        Saves the audio returned by the pipeline to files.
 
-        output_dict["audio"] = waveform.to(device="cpu", dtype=torch.float).numpy()
-        output_dict["sampling_rate"] = self.sampling_rate
+        Args:
+            audio (`dict[str, Any]` or `list[dict[str, Any]]`):
+                The audio returned by the pipeline. The dictionary (or each dictionary, if it is a list) should
+                contain two keys:
+                - `"audio"`: The audio waveform.
```

Comment:
Cool to have saving function 👍 

Maybe useful to mention that the audio is expected to have shape `(nb_channels, audio_length)`? (unless self.processor expects something else...) 

And are we sure it isn't a tensor? Otherwise can also add something like [this](https://github.com/huggingface/transformers/blob/9b3203f47b5cabb5c318ce1458438847df23fb1d/src/transformers/models/dia/processing_dia.py#L407)?

I guess `synthesiser` and `tts_pipeline` try to ensure this dimension order, but maybe useful in case people gets issues saving.